src/merkle/store/mod.rs

```diff
@@ -108,22 +113,45 @@ impl MerkleStore {
         Ok(hash.into())
     }
 
-    /// Returns the path for the node at `index` rooted on the tree `root`.
+    /// Returns the leaf value at `index` and its opening to `root`.
```

Comment:
Should this be "leaf value"? Technically we would be returning any node value at the specified index. So, maybe a better phrasing would be "Returns the node at the specified `index` and its opening to the `root`"?

Reply:
changed from `leaf` to `node`